blockchain/types/accountkey/account_key_serializer.go

```diff
@@ -72,17 +77,20 @@ func (serializer *AccountKeySerializer) MarshalJSON() ([]byte, error) {
 		return nil, err
 	}
 
-	return json.Marshal(&accountKeyJSON{serializer.keyType, b})
+	return json.Marshal(&AccountKeyJSON{&serializer.keyType, b})
 }
 
 func (serializer *AccountKeySerializer) UnmarshalJSON(b []byte) error {
-	var keyJSON accountKeyJSON
+	var keyJSON AccountKeyJSON
 
 	if err := json.Unmarshal(b, &keyJSON); err != nil {
 		return err
 	}
 
-	serializer.keyType = keyJSON.KeyType
+	if keyJSON.KeyType == nil {
+		return errNoKeyType
+	}
+	serializer.keyType = *keyJSON.KeyType
```

Comment:
During the JSON unmarshalling, `KeyType` could be assigned to `uint8(0)` when `KeyType` was not specified on the JSON input. I thought this behavior doesn't give any tolerance to users' mistakes.

Suppose users/developers wanted to unmarshal complicated `AccountKeyRoleBased`, but forgot to specify `KeyType`. The unmarshalling will return `AccountKeyNil`, which means `uint(0) KeyType`, without any errors. 

To more informative to users, I changed the type of `KeyType` to a pointer and added `nil` check during the JSON unmarshalling.